vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/retrievable/Retrievable.kt

```diff
@@ -61,4 +66,25 @@ interface Retrievable : Persistable {
      * @return [RetrievableAttribute] or null.
      */
     fun <T : RetrievableAttribute> filteredAttribute(c: Class<T>): T?
+
+    /**
+     * Adds a [Relationship] to this [Retrievable].
+     *
+     * @param relationship The [Relationship] to add.
+     * @return True on success, false otherwise.
+     */
+    fun addRelationship(relationship: Relationship): Boolean
+
+    /**
+     * Removes all [Relationship]s from this [Retrievable].
+     *
+     * @param relationship The [Relationship] to remove.
+     * @return True on success, false otherwise.
+     */
+    fun removeRelationship(relationship: Relationship): Boolean
+
+    /**
+     * Finds all [Relationship]s held by this [Retrievable] that satisfy the given [Predicate]
+     */
+    fun findRelationship(predicate: Predicate<Relationship>): List<Relationship> = this.relationships.filter { predicate.test(it) }
```

Comment:
What would be more idiomatic?

Reply:
Having a Kotlin lambda rather than a Java function interface. Or are they mapped correctly? Can you call
```kotlin

findRelationship{
 ...
}

```
?

Reply:
Yes you can!